Lab1/src/FirstSteps.hs

```diff
@@ -17,9 +17,11 @@ xor x y = error "todo"
 -- median3 1 3 2 == 2
 -- median3 5 2 5 == 5
 max3, median3 :: Integer -> Integer -> Integer -> Integer
-max3 x y z = error "todo"
-
-median3 x y z = error "todo"
+max3 x y z = max (max x y) z  -- if z >= a then z else a where {a = x >= y then z else y} 
+median3 x y z | (min x z <= y) && (y <= max x z) = y --sum
+              | (min y z <= x) && (x <= max y z) = x
+              | (min x y <= z) && (z <= max x y) = z
+              | otherwise = error "the median has not been calculated"
```

Comment:
Возможный ли это случай?

Reply:
По идее, первые три условия покрывают все возможные случаи. Последний был приведен по большей части для сохранения стиля написания охранных условий (насколько я понял, otherwise прописывается в большинстве случаев).

Reply:
Да, но поэтому можно уже третий случай сделать `otherwise`.